generator.rb

```diff
@@ -0,0 +1,85 @@
+def is_vowel?(letter)
+    letter.each_char do |char|
+        if char == "a" || char == "e" || char == "i" || char == "o" || char == "u"
+            return true
+        else
+            return false
+        end
+    end
+end
+
+def input_validation(word)
+    while word.length < 2 do
+        puts "The word must have 2 or more letters. Please enter again."
+        word = gets.chomp
+    end
+    
+    while !(word.include?('a') || word.include?('e') || word.include?('i') || word.include?('o') || word.include?('u')) do
+        puts "Input is invalid. The word needs at least 1 vowel. Try again."
+        word = gets.chomp
+    end
+    return word
```

Comment:
There is a bug here! It never checks if new input has at 2 or more letters again. To recreate: run `ruby generator.rb` to run program. When asked for the first input, type `xx`. It will say `Input is invalid. The word needs at least 1 vowel. Try again.`. Then, type in `a`. It will accept `a` and then move on and ask `What is the second word to combine?`. That's because the `while` loop that checks if there's a vowel does not ALSO check if the word length is valid again!

Reply:
Resolved the bug and combined the two while loops into one.